test/api/advanced/find-or-create.test.js

```diff
@@ -151,3 +151,76 @@ test('advanced.findOrCreate (protected metadata fields)', (t) => {
     )
   })
 })
+
+test('advanced.findOrCreate (encryption - GroupId)', (t) => {
+  const sbot = Testbot()
+
+  const groupId = '%EPdhGFkWxLn2k7kzthIddA8yqdX8VwjmhmTes0gMMqE=.cloaked'
+  const groupKey = Buffer.from(
+    '30720d8f9cbf37f6d7062826f6decac93e308060a8aaaa77e6a4747f40ee1a76',
+    'hex'
+  )
+  sbot.box2.addGroupKey(groupId, groupKey)
+
+  testReadAndPersisted(t, sbot, (t, sbot, cb) => {
+    sbot.metafeeds.advanced.findOrCreate((err, mf) => {
+      if (err) t.error(err, 'no error')
+
+      sbot.metafeeds.advanced.findOrCreate(
+        mf,
+        (f) => f.feedpurpose === 'private',
+        {
+          feedpurpose: 'private',
+          feedformat: 'classic',
+          recps: [groupId],
+          encryptionFormat: 'box2'
+        },
+        (err, f) => {
+          if (err) t.error(err, 'no error')
+
+          t.deepEqual(f.recps, [groupId], 'FeedDetails contains recps')
+
+          sbot.db.query(where(type('metafeed/add/derived')), toCallback((err, msgs) => {
+            if (err) return cb(err)
+
+            if (msgs.length !== 1) t.equal(msgs.length, 1, 'only one metafeed/add/derived')
```

Comment:
I'm confused, isn't there supposed to be 2 `metafeed/add/derived` msgs? 1 for the root mf adding a shard, and 1 for the shard feed adding the group feed.

Reply:
Also, why the if condition?

```suggestion
            t.equal(msgs.length, 1, 'only one metafeed/add/derived')
```

Reply:
Good question - I think this is a mistake in the tests to only expect 1. I wonder if it's passing because of a race condition around indexing (sigh).

I'm noticing this is quite common (if it's what it is) - to the point that I want a helper function. e.g.

```js
ssb.db.indexesReady(cb)
```

will investigate and report back

Reply:
For jitdb, everything you query should be up-to-date with the log. But for leveldb indexes, there is `ssb.db.onDrain(nameOfTheLevelIndex, cb)` as a way of waiting until it's indexed

Reply:
OOOOH, the reason there's only 1 `metafeed/add/derived` is because this is testing the lower level (advanced) API which doesn't do sharding